core-web/apps/dotcms-ui/src/app/portlets/dot-categories/dot-categories-create-edit/dot-categories-properties/store/dot-categories-properties.store.ts

```diff
@@ -0,0 +1,110 @@
+import { Injectable } from '@angular/core';
+import { ComponentStore } from '@ngrx/component-store';
+import { Observable, of } from 'rxjs';
+import { catchError, switchMap, tap } from 'rxjs/operators';
+import { HttpErrorResponse } from '@angular/common/http';
+
+import { DotMessageService } from '@services/dot-message/dot-messages.service';
+import { DotHttpErrorManagerService } from '@dotcms/app/api/services/dot-http-error-manager/dot-http-error-manager.service';
+import { DotRouterService } from '@services/dot-router/dot-router.service';
+import { DotCategoriesUtillService } from '@dotcms/app/api/services/dot-categories/dot-categories-utill.service';
+import { DotGlobalMessageService } from '@components/_common/dot-global-message/dot-global-message.service';
+
+import {
+    DotCategory,
+    DotCategoryPayload
+} from '@dotcms/app/shared/models/dot-categories/dot-categories.model';
+
+export interface DotCategoriesPropertiesState {
+    category: DotCategory;
+    apiLink: string;
+}
+
+@Injectable()
+export class DotCategoriesPropertiesStore extends ComponentStore<DotCategoriesPropertiesState> {
+    constructor(
+        private dotMessageService: DotMessageService,
+        private dotGlobalMessageService: DotGlobalMessageService,
+        private dotCategoriesUtillService: DotCategoriesUtillService,
+        private dotHttpErrorManagerService: DotHttpErrorManagerService,
+        private dotRouterService: DotRouterService
+    ) {
+        super({
+            category: null,
+            apiLink: ''
+        });
+    }
+
+    readonly vm$ = this.select((state: DotCategoriesPropertiesState) => {
+        return state;
+    });
+
+    readonly categoryAndStructure$ = this.select(({ category }: DotCategoriesPropertiesState) => {
+        return {
+            category
+        };
+    });
+
+    readonly updateIsContentTypeButtonEnabled = this.updater<boolean>(
+        (state: DotCategoriesPropertiesState, isContentTypeButtonEnabled: boolean) => {
+            return {
+                ...state,
+                isContentTypeButtonEnabled
```

Comment:
`isContentTypeButtonEnabled` doesn't exist in `DotCategoriesPropertiesState` interface. What you return here is the state.